### ISSUE TITLE
CanuTrio de-duplication

### ISSUE THREAD

**Issue by pcarbone** [marbl/canu#1672]
Hi,
I have two question regarding deduplication of CanuTrio assemblies.
I have haplotype assemblies after CanuTrio that according to BUSCO are complete (97%) but 15% duplicated.
It seems that duplication is due to the presence of contigs belonging to the other parental haplotype. Duplication is efficiently reduced by running purge_dups. However, as purge_dups and other deduplication tools are not genotype-aware, they are indiscriminately removing contigs from one or the other haplotype on each haplotype assembly.

Is there a way to reduce duplication for CanuTrio assemblies in a haplotype-specific manner?

For deduplication of each haplotype assembly I also tried to run canu -haplotype (version 1.8). I used contigs as corrected reads input for their binning in the two haplotypes according to kmers in short reads of the parents.
Somehow surprisingly, 5% of each haplotype assembly was still binned to the other parental haplotype (data below). BUSCO completeness and duplication are decreased by the same extent (by 2%) in the assembly fraction in which contigs binned to the other parental haplotype are removed. So this contig binning is not efficiently deduplicating either.

Does it make sense that CanuTrio still finds full contigs that can be binned to the other haplotype if the assembly was done with haplotype-binned reads already?
Any other recommendation to get rid of duplication?

Thanks.

```
-- Loading haplotype data.
--  Haplotype './0-kmers/haplotype-A.meryl':
--   use kmers with frequency at least 20.
--   loaded 31188590 kmers.
--  Haplotype './0-kmers/haplotype-B.meryl':
--   use kmers with frequency at least 15.
--   loaded 46741865 kmers.
-- Data loaded.
--
-- Processing reads in batches of 100 reads each.
--
--      188 reads     23790527 bases written to haplotype file ./haplotype-A.fasta.gz.
--      804 reads    567347243 bases written to haplotype file ./haplotype-B.fasta.gz.
--       12 reads       584689 bases written to haplotype file ./haplotype-unknown.fasta.gz.
```

**Comment by skoren** [comment]
Do both bins have a similar BUSCO duplication? We've seen some cases where there's a "ghosting", especially of the sex chromosomes for F1 whose parents were closely related. However, these are typically short contigs and shouldn't contain that high of a duplication fraction. I'd think a likelier explanation is that there really is duplication in the genome.

I'd suggest using merqury for validation, it will provide phase blocks on each contig, you can look for contigs with a higher fraction of the alternate  haplotype counts and look at the blob plots.

**Comment by pcarbone** [comment]
Dear Sergey, than you for your prompt reply.

Both bins show similar 12-15% duplication. This is CanuTrio assembly for a plant outbred (1.5% het in GenomeScope from short reads).
It seems duplication is an assembly artefact because:
- Each bin assembly size is 10-15% larger than the estimated haploid genome size. 
- Duplicated contigs match the genotype of the alternate bin.
- Purge_dups efficiently reduce duplication to only 3-5%. The problem is this deduplication is not bin-genotype-aware, which leads to poorer genome binning after deduplication.

I did not try Mercury yet. I will.
Thanks.

**Comment by skoren** [comment]
Any updates, were you able to run merqury to identify markers in contigs and which to remove?

**Comment by pcarbone** [comment]
I struggled a bit converting merqury Slurm scripts to our SGE system and I ran an alternative script at the end.
I removed contigs that show high similarity (dv <0.05) to the alternate haplotype and are >75% covered by another contig in the current haplotype. Then I ran purge_dups on the alt sim deduplicated assembly to further purge contigs duplicated within the current haplotype itself. After that, assembly is close to the expected size, BUSCO completeness is kept and BUSCO duplication drops from 12-15% to only 2-3%, which could more reliably be in the range of real biological duplication.
